test/test_code.py

```diff
@@ -313,86 +296,119 @@ def scrape(url):
     assert len(codeblocks) == 1 and codeblocks[0] == ("", "source setup.sh")
 
 
```

Comment:
why remove the `in_docker_container` checks here?

Reply:
My thinking was, technically, you could have docker inside a docker container. Also, we want to move toward the direction of always separating code execution environment from the application environment, hence the test set up this way. This PR added `--skip-docker` test so for someone developing inside docker, without a nested docker setup, they can use that flag to skip tests with `use_docker=True`.

Reply:
Ideally the next step would be to use a separate docker container for code execution, and communicate with the application via websocket.

Reply:
"always separating code execution environment from the application environment" is questionable. We have examples in https://github.com/microsoft/autogen/blob/main/notebook/agentchat_auto_feedback_from_code_execution.ipynb that executes the code in the same Jupyter notebook.

Reply:
That's one usage scenario. I think the original question from @olgavrou  was why not turning off the test when it is in docker. On that I said we can turning off the test by setting `--skip-docker`, as we don't want to assume people developing inside docker always want to skip these test. 

If you guys agree, maybe we should resolve this.